internal/utils/statefulset.go

```diff
@@ -51,3 +55,35 @@ func GetStatefulSet(ctx context.Context, cl client.Client, etcd *druidv1alpha1.E
 
 	return nil, nil
 }
+
+// FetchPVCWarningMessagesForStatefulSet fetches warning messages for PVCs for a statefulset, if found concatenates the first 2 warning messages and returns
```

Comment:
Again just for my understanding why we need the first 2 messages only?
May be explaining the rationale in the comment will keep the context preserved for future code changes.

Reply:
Again, this was just copied over from the previous code base. Fetching older events could lead to recording a wrong condition for the etcd status, since the event may have been generated for an older sts version, and was later rectified. Ideally, just the latest event about the PVC is sufficient, but I think we had added a buffer of one more event, just in case.

If you have no string objection to this, we can leave it as is, and observe how it works for real PVC failures and then re-assess this if necessary.

Reply:
Yes, no strong objection. All Good.